p2p_wallet/Services/NotificationsService/NotificationService.swift

```diff
@@ -1,22 +1,64 @@
 //
-//  NotificationsService.swift
+//  NotificationService.swift
 //  p2p_wallet
 //
 //  Created by Chung Tran on 22/12/2021.
 //
 
 import Foundation
+import Resolver
 import UIKit
 
-protocol NotificationsServiceType {
+protocol NotificationService {
+    typealias DeviceTokenResponse = JsonRpcResponseDto<DeviceTokenResponseDto>
+
+    func sendRegisteredDeviceToken(_ deviceToken: Data)
     func showInAppNotification(_ notification: InAppNotification)
+    func wasAppLaunchedFromPush(launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool
 }
 
-class NotificationsService: NotificationsServiceType {
+final class NotificationServiceImpl: NotificationService {
+    @Injected private var accountStorage: AccountStorageType
+    @Injected private var notificationRepository: NotificationRepository
+
+    private let deviceTokenKey = "deviceToken"
+
+    init() {
+        guard let deviceToken = UserDefaults.standard.data(forKey: deviceTokenKey) else { return }
+        sendRegisteredDeviceToken(deviceToken)
+    }
+
+    func sendRegisteredDeviceToken(_ deviceToken: Data) {
+        guard let publicKey = accountStorage.account?.publicKey.base58EncodedString else { return }
+        let tokenParts = deviceToken.map { data in String(format: "%02.2hhx", data) }
+        let token = tokenParts.joined()
+
+        Task {
```

Comment:
Make function async and wrap it in Task outside will be better.

It indicates that the function of service is async in nature. It's also let caller to wait on the function if it's required.